applications/callflow/src/cf_endpoint.erl

```diff
@@ -116,6 +118,14 @@ maybe_have_endpoint(JObj, EndpointId, AccountDb) ->
             has_endpoint(JObj, EndpointId, AccountDb, EndpointType)
     end.
 
+-spec maybe_map_endpoint_type(api_binary()) -> api_binary().
+maybe_map_endpoint_type('undefined') -> 'undefined';
+maybe_map_endpoint_type(EndpointType) ->
```

Comment:
Was there a reason why you chose a proplist versus using pattern matching in the function clauses?

```erlang
maybe_map('undefined') -> 'undefined';
maybe_map(<<"clicktocall">>) -> kz_device:type();
maybe_map(Type) -> Type.
```